src/watch-mode-spy.test.ts

```diff
@@ -0,0 +1,65 @@
+import WatchModeSpy from './watch-mode-spy';
+
+describe('watchModeSpy', () => {
```

Comment:
Regarding writing integration tests, you could add such integration tests in jest, if you feel like it. It would involve:

1. Creating a new test temporary directory
2. Starting a new `tsc` process in watch mode

and then, depending on the level of integration you want the tests to be at, you could either:

1. Read the stdout of that `tsc` process and invoke corresponding `WatchModeSpy` methods, making assertions such as:
    1. Initially, it should `detectWatchModeCompilationStartOutput` and `detectWatchModeCompilationFinishedOutput`
    2. After the code modifies a file in the filesystem, it should `detectWatchModeFileChangeDetectedOutput` and then `detectWatchModeCompilationFinishedOutput`
    
    This sounds testable. It tests `WatchModeSpy` kind of in isolation from the rest of the program, but that is ok. It gives us confidence that this part works, and the rest of the program is tested with existing integration tests.

2. Pass the output directly to `program` and make assertions about the `program`'s output

    This is more involved, gives us more confidence that the entire tool works end-to-end. If there are problems, it will be very coarse and not give us precise signs where the root of the problem is, just that something is broken.

I believe testing `WatchModeSpy` in isolation against the output of `tsc -w` is good enough.

It is also fine if you want to skip adding these tests in this PR. They can be added later, either by me or by you, or by some other contributor